core/src/main/java/org/frankframework/pipes/ForPipe.java

```diff
@@ -115,26 +117,29 @@ private Integer determineStopAtValue(Message message, PipeLineSession session) t
 			return stopAt;
 		}
 
-		ParameterValueList parameterValueList = getParameterValueList(message, session);
-		ParameterValue stopAtParameter = parameterValueList.get(STOP_AT_PARAMETER_VALUE);
+		Optional<ParameterValueList> optionalParameterValueList = getParameterValueList(message, session);
+
+		if (optionalParameterValueList.isPresent()) {
+			ParameterValue stopAtParameter = optionalParameterValueList.get().get(STOP_AT_PARAMETER_VALUE);
 
-		if (stopAtParameter.getValue() == null || StringUtils.isBlank(stopAtParameter.asStringValue())) {
-			throw new PipeRunException(this, "Can't determine 'stopAt' value");
+			if (stopAtParameter.getValue() != null && StringUtils.isNotBlank(stopAtParameter.asStringValue())) {
+				return stopAtParameter.asIntegerValue(0);
+			}
 		}
 
-		return stopAtParameter.asIntegerValue(0);
+		throw new PipeRunException(this, "Can't determine 'stopAt' value");
 	}
 
-	private ParameterValueList getParameterValueList(Message message, PipeLineSession session) throws PipeRunException {
+	private Optional<ParameterValueList> getParameterValueList(Message message, PipeLineSession session) throws PipeRunException {
 		if (getParameterList() != null) {
 			try {
-				return getParameterList().getValues(message, session);
+				return Optional.of(getParameterList().getValues(message, session));
 			} catch (ParameterException e) {
 				throw new PipeRunException(this, "exception extracting parameters", e);
 			}
 		}
 
-		return null;
+		return Optional.empty();
```

Comment:
Kunnen we niet een lege ParameterValueList teruggeven? Dat lijkt mij schonere code geven met minder extra condities en checks in de aanroep.

Reply:
Vast, maar dan gaan we denk ik wel een beetje ver voor dit ticket. Je zou dan eigenlijk ook gelijk `AbstractPipe.getParameterList()` willen aanpassen naar een lege lijst ipv null, aangezien je daarop `getValues()` aanroept. Wat vind je ervan als ik daar een apart ticket voor maak?

Reply:
Dat lijkt me prima.